js/flexbuffers.js

```diff
@@ -0,0 +1,1015 @@
+      isFloat: function() { return flexbuffers.ValueType.FLOAT === valueType || flexbuffers.ValueType.INDIRECT_FLOAT === valueType; },
+      isInt: function() { return this.isNumber() && !this.isFloat(); },
+      isString: function() { return flexbuffers.ValueType.STRING === valueType || flexbuffers.ValueType.KEY === valueType; },
+      isBool: function() { return flexbuffers.ValueType.BOOL === valueType; },
+      isBlob: function() { return flexbuffers.ValueType.BLOB === valueType; },
+      isVector: function() { return flexbuffers.ValueType.isAVector(valueType); },
+      isMap: function() { return flexbuffers.ValueType.MAP === valueType; },
+
+      boolValue: function() {
+        if (this.isBool()) {
+          return readInt(dataView, offset, parentWidth) > 0;
+        }
+        return null;
+      },
+
+      intValue: function() {
+        if (valueType === flexbuffers.ValueType.INT) {
+          return readInt(dataView, offset, parentWidth);
+        }
+        if (valueType === flexbuffers.ValueType.UINT) {
+          return readUInt(dataView, offset, parentWidth);
+        }
+        if (valueType === flexbuffers.ValueType.INDIRECT_INT) {
+          return readInt(dataView, indirect(dataView, offset, parentWidth), flexbuffers.BitWidth.fromByteWidth(byteWidth));
+        }
+        if (valueType === flexbuffers.ValueType.INDIRECT_UINT) {
+          return readUInt(dataView, indirect(dataView, offset, parentWidth), flexbuffers.BitWidth.fromByteWidth(byteWidth));
+        }
+        return null;
+      },
+
+      floatValue: function() {
+        if (valueType === flexbuffers.ValueType.FLOAT) {
+          return readFloat(dataView, offset, parentWidth);
+        }
+        if (valueType === flexbuffers.ValueType.INDIRECT_FLOAT) {
+          return readFloat(dataView, indirect(dataView, offset, parentWidth), flexbuffers.BitWidth.fromByteWidth(byteWidth));
+        }
+        return null;
+      },
+
+      numericValue: function() { return this.floatValue() || this.intValue()},
+
+      stringValue: function() {
+        if (valueType === flexbuffers.ValueType.STRING || valueType === flexbuffers.ValueType.KEY) {
+          const begin = indirect(dataView, offset, parentWidth);
+          return fromUTF8Array(new Uint8Array(dataView.buffer, begin, this.length()));
+        }
+        return null;
+      },
+
+      blobValue: function() {
+        if (this.isBlob()) {
+          const begin = indirect(dataView, offset, parentWidth);
+          return new Uint8Array(dataView.buffer, begin, this.length());
+        }
+        return null;
+      },
+
+      get: function(key) {
+        const length = this.length();
+        if (Number.isInteger(key) && flexbuffers.ValueType.isAVector(valueType)) {
+          if (key >= length || key < 0) {
+            throw `Key: [${key}] is ot applicable on ${path} of ${valueType} length: ${length}`;
```

Comment:
This changes behavior compared to the other implementations. Note that in C++ no operation returns an error or raises an exception, it returns Null references for things that don't exist or where the conversion doesn't make sense. This makes code more robust against type changes. Please follow that same path in JS too. This code should ideally be free of any `throw`s

Reply:
I copied this behaviour from the Dart implementation. In Dart I introduced the exception throwing and a path variable after some discussions in PR:

https://github.com/google/flatbuffers/pull/5853#discussion_r421636050

If this is not desirable in JavaScript I can remove the path property from the `Reference` and return `null` instead of throwing an exception.